pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -33,145 +33,72 @@ type StorageCohort struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec defines the behavior of a cohort.
+	// +optional
```

Comment:
How is this optional?

Reply:
I thought to keep all these optional following the k8s pattern. Kubernetes also keeps the spec section of resources optional. So that way. Will change in our case to make it mandatory.

Reply:
If we add optional tag then open api spec validation will be removed at the time of automated CRD yaml generation. We should keep it.

Reply:
Let say we are using kubectl to apply a cr if in the crd validation is present then open api validation is done by kube api server itself. And crd yaml dependent on these annotation.